protocols/request-response/src/cbor.rs

```diff
@@ -130,24 +133,43 @@ mod codec {
         where
             T: AsyncWrite + Unpin + Send,
         {
-            let data: Vec<u8> = serde_cbor::to_vec(&resp).map_err(into_io_error).unwrap();
+            let data: Vec<u8> =
+                cbor4ii::serde::to_vec(Vec::new(), &resp).map_err(IntoError::into_io_error)?;
 
             io.write_all(data.as_ref()).await?;
 
             Ok(())
         }
     }
 
-    fn into_io_error(err: serde_cbor::Error) -> io::Error {
-        if err.is_syntax() || err.is_data() {
-            return io::Error::new(io::ErrorKind::InvalidData, err);
-        }
+    trait IntoError {
+        fn into_io_error(self) -> io::Error;
+    }
```

Comment:
I'd suggest we remove this trait and have two functions:

- `decode_error_into_io`
- `encode_error_into_io`

Reply:
This is what I initially had haha. The names are a bit iffy, but easy enough.